Lib/pathlib.py

```diff
@@ -853,32 +868,16 @@ class PureWindowsPath(PurePath):
 # Filesystem-accessing classes
 
 
-class Path(PurePath):
-    """PurePath subclass that can make system calls.
+class _AbstractPath(PurePath):
+    """PurePath subclass with abstract methods for making system calls.
```

Comment:
Have you thought about making the class a formal ABC (subclass of abc.ABC + abstractmethod decorators)?

Reply:
I considered it, but I don't think we should require users to implement all abstract methods. pathlib.Path itself can already raise NotImplementedError from readlink(), symlink_to(), hardlink_to(), owner() and group(), so would fall afoul of the requirement.

Reply:
To me, that sounds like a reason in _favour_ of having this being an ABC — it could help clarify the interface. You could add `abstractmethod` to the "core" methods which _do_ need to be implemented in all subclasses, and leave the other quasi-abstract-methods as normal methods that raise `NotImplementedError`. Then it would be clear to users of this class which methods are essential to implement in subclasses, and which are merely optional.

Reply:
Gotcha! I'll give it some thought. The group of "core" methods might be rather small: perhaps only `stat()`, `open()` and `iterdir()`. which would cover the needs of `importlib.abc.Traversable`.

Reply:
I've now adjusted the code to subclass from `abc.ABC`, and decorated the core methods with `@abc.abstractmethod`

But I'm a little spooked by this related code block:

```python
# Can't subclass os.PathLike from PurePath and keep the constructor
# optimizations in PurePath._parse_args().
os.PathLike.register(PurePath)
```

This implies to be that messing with the metaclass, e.g. via inheriting `abc.ABC` either directly or via `os.PathLike`, causes issues in `_parse_args()`. But I can't see any from local testing! All the same branches seem to be taken, and the test suite passes.

In my last commit I've removed the above code block and made `PurePath` inherit from `PathLike`. If this causes problems, then inheriting from `ABC` in `_AbstractPath` will also, I think.